examples/src/main/scala/caliban/ExampleData.scala

```diff
@@ -22,18 +25,50 @@ object ExampleData {
     case class Mechanic(shipName: String) extends Role
   }
 
-  case class Character(name: String, nicknames: List[String], origin: Origin, role: Option[Role])
+  @GQLDirective(federation.Key("name"))
```

Comment:
I wonder if all this might confuse people who are just looking for the regular Caliban example. Maybe we could have a dedicated data and API example for federation?

Reply:
Yeah I don't mind, I was worried about duplicating code and then having drift, but since these are examples that doesn't matter too much.